bin/bench-tx/bench-tx.json

```diff
@@ -1,23 +1,23 @@
 {
   "simple": {
-    "prologue": 4167,
-    "notes_processing": 2540,
+    "prologue": 4221,
+    "notes_processing": 2652,
     "note_execution": {
-      "0x7fe47b84889353de93aea4bfeaa8da5da0036ea7a84dfc5ff5765ec51440896f": 1266,
-      "0xbf1c92f4f0b35b51d3f3e8f7ba19abf7718a7e087627df0d17987af8a1cb872a": 1233
+      "0x2a96867a0f22d54d1e3530c074c3ad9ba9542b5060edf2f8398202f5d2e3f557": 1322,
+      "0xc52584ba513fd5eaaec47234263a1b20ce4d848488430e577dbdd6d9254d5bae": 1289
     },
     "tx_script_processing": 45,
-    "epilogue": 1525,
-    "after_compute_fee_cycles": 1143
+    "epilogue": 2447,
```

Comment:
Question: why did the epilogue cycle count increased by this much here?

Reply:
I'm pretty sure it wasn't caused by this PR or a recent one. I can try to bisect search it tomorrow.

Reply:
Ah - interesting! Thank you for looking into this.

I guess I didn't realize that removing an asset from the vault would take up roughly 1K cycles. Could you check if the main driver of this are the `link_map` operations? I believe we are doing both `link_map::get` and `link_map::set` here.

Reply:
Makes sense! Thank you! I don't think we need to investigate more now, but would be good to create an issue so that we could investigate more in the future. There are probably ways to make it more efficient by relying more on the host - but not a very high priority for now.

Reply:
btw, I would make the issue more about optimizing add/remove assets than about the link map specifically.

Reply:
Created https://github.com/0xMiden/miden-base/issues/1715.